README.md

```diff
@@ -309,6 +286,34 @@ Kubeclient::Client.new(
 )
 ```
 
+
+#### Google's Application Default Credentials
+
+On Google Compute Engine, Google App Engine, or Google Cloud Functions, as well as `gcloud`-configured systems
+with [application default credentials](https://developers.google.com/identity/protocols/application-default-credentials),
+kubeclient can use `googleauth` gem to authorize.
+
+This requires the [`googleauth` gem](https://github.com/google/google-auth-library-ruby) that is _not_ included in
+`kubeclient` dependencies so you should add it to your bundle.
+
+If you use `Config.context(...).auth_options` and the kubeconfig file has `user: {auth-provider: {name: gcp}}`, kubeclient will automatically try this (raising LoadError if you don't have `googleauth`).
+
+Or you can obtain a token manually:
+
+```ruby
+require 'googleauth'
+
+auth_options = {
+  bearer_token: Kubeclient::GoogleApplicationDefaultCredentials.token
+}
+client = Kubeclient::Client.new(
+  'https://localhost:8443/api/', 'v1', auth_options: auth_options
+)
+```
+
+Note that this returns token good for one hour. If your code runs for longer than that, you should plan to
+acquire a new one.
```

Comment:
Q: how do you acquire a new one, in the automagic Config `gcp` scenario?
Calling `.context` again will do this, right?

Reply:
Call `Kubeclient::GoogleApplicationDefaultCredentials.token` again.

I think we've discussed before that `kubeclient` doesn't have any support for renewing tokens. Is that something that's now covered with `exec` functionality?

Reply:
Calling `.context` will generate a new token, since it will end up calling `Kubeclient::GoogleApplicationDefaultCredentials.token`, as Jeremy points out. I think the farthest we should go in that direction is to capture the error, if possible, and let the caller handle it.